pkg/universe.dagger.io/x/berryphillips@gmail.com/onepassword/read.cue

```diff
@@ -0,0 +1,55 @@
+package onepassword
+
+import (
+	"dagger.io/dagger/core"
+	"universe.dagger.io/docker"
+)
+
+// Read a secret by secret reference
+#Read: {
+	// The 1Password account login credentials
+	credentials: #Credentials
+
+	// The reference to the 1Password secret
+	reference: #Reference
+
+	_base: docker.#Pull & {
+		source: "1password/op:\(#_Version)"
+	}
+
+	_container: docker.#Run & {
+		input: _base.output
+
+		command: {
+			name: "bash"
+			flags: "-c": "/scripts/read.sh > /tmp/secret"
+		}
+
+		_scripts: core.#Source & {
+			path: "."
+			include: ["*.sh"]
+		}
+
+		mounts: scripts: {
+			contents: _scripts.output
+			dest:     "/scripts"
+		}
+
+		env: {
+			ADDRESS:    credentials.address
+			EMAIL:      credentials.email
+			SECRET_KEY: credentials.secretKey
+			PASSWORD:   credentials.password
+			REFERENCE:  reference.output
+		}
+
+		always: true
+	}
+
+	_secret: core.#NewSecret & {
+		input: _container.output.rootfs
+		path:  "/tmp/secret"
+	}
```

Comment:
I just figure out that we could maybe also add the possibility to export secret from `docker.#Run`, WDYT @shykes @helderco ?

Reply:
Makes sense to me, I vaguely remember that we discussed it at some point, but don't remember what the conclusion was.

Reply:
Same logic as exporting a file or directory so makes sense.

Reply:
@TomChv Updated to use the new secrets export feature!